src/utils/actions.js

```diff
@@ -0,0 +1,254 @@
+          const searchParam = dateFilter.searchParam;
+          const filterPeriod = dateFilter.valuePeriod; // assume its a valuePeriod
+          if (path) {
+            const dateFiltered = resources.filter(resource => {
+              // get value by path
+              const date = getByPath(resource, path); // assume its a datetime?
+              return compareDates(date, filterPeriod);
+            });
+            filteredResources.push.apply(filteredResources, dateFiltered);
+          } else {
+            // path or searchParam are required
+            const dateFiltered = resources.filter(resource => {
+              // get value by param
+              const date = getBySearchParam(resource, searchParam); // assume its a datetime?
+              return compareDates(date, filterPeriod);
+            });
+            filteredResources.push.apply(filteredResources, dateFiltered);
+          }
+        } else {
+          // if there's no date filter, we can just push all the resources
+          // in to be filtered by code
+          filteredResources.push.apply(filteredResources, resources);
+        }
+
+        const codeFilter = input.codeFilter;
+        if (codeFilter) {
+          // code filter isn't under the same constraint as datefilter
+          // we can assume that it's just a path
+          const path = codefilter.path;
+          const searchParam = codeFilter.searchParam;
+          if (path) {
+            // no need to refilter resources
+            const filteredResources = filteredResources.filter(resource => {
+              const code = getByPath(resource, path);
+              return compareCodes(code, codeFilter.code); // codeFilter.code is of type Coding, could also be a valueSet?
+            });
+          } else if (searchParam) {
+            const filteredResources = filteredResources.filter(resource => {
+              const code = getBySearchParam(resource, searchParam);
+              return compareCodes(code, codeFilter.code);
+            });
+          }
+        }
+        return context.records.push(filteredResources);
+      });
+    }
+  },
+  'execute-research-query': context => {
+    // noop
+  }
+};
+
+function validateProfile(report, structureDefinition) {
+  // TODO: Perhaps this is something one of the servers/clients would handle?
+  return report;
+}
+
+function getByPath(resource, path) {
+  // TODO: function to search resources by path
+  // and return correct attribute
+  return resource;
+}
+
+function compareCodes(coding1, coding2) {
+  // assume they're both coding
+  return coding1.code === coding2.code && coding1.system === coding2.system;
+}
+
+function getBySearchParam(resource, searchParam) {
+  // TODO: function to search resources by param
+  // and return correct attribute
+  return resource.date;
+}
+
+function compareDates(period1, period2) {
+  // TODO: returns true if the periods overlap
+}
+function createBundle(records) {
```

Comment:
Can we make this conformant to the two bundle profiles (shouldn't be difficult based on the differential table)
[Public Health Reporting Bundle](https://build.fhir.org/ig/HL7/fhir-medmorph/StructureDefinition-us-ph-reporting-bundle.html)
[Public Health Content Bundle](https://build.fhir.org/ig/HL7/fhir-medmorph/StructureDefinition-us-ph-content-bundle.html)

Reply:
It seems like the only change to be made here is the type needs to be `collection` for the content bundle